1_dynamic/future.k

```diff
@@ -241,28 +247,68 @@ module FUTURE
 */
 
   // the case when the future is not done
-  rule (<active-task> Task
-           <k> future(Id).get ~> K </k>
-        </active-task> => <active-task>... <futureId> -1 </futureId> <k> . </k> ...</active-task> )
-       (. => <waiting-task> <guard> future(Id).get ~> K </guard>
-                              <waiting-task-conf> Task </waiting-task-conf>
+  rule ( <active-task> Task
+           <k> waiting(<guard> Guards:Set </guard>) ~> K </k>
+         </active-task>
+        =>
+         <active-task>...
+           <futureId> -1 </futureId>
+           <k> . </k>
+         ...</active-task> )
+       (. => <waiting-task>
+               <guard> Guards </guard>
+               <waiting-k> K </waiting-k>
+               <waiting-task-conf> Task </waiting-task-conf>
              </waiting-task>)
-       <fid> Id </fid>
-       <state> S:State </state>
-       requires S =/=K completed
-  [futureBlock]
 
   // the case when the future is done
   rule (<active-task>... <futureId> -1 </futureId> ...</active-task>
         =>
-        <active-task> Task <k> future(Id).get ~> K </k> </active-task>
+        <active-task> Task <k> K </k> </active-task>
        )
-       (<waiting-task> <guard> future(Id).get ~> K </guard>
-                         <waiting-task-conf> Task </waiting-task-conf>
+       (<waiting-task> <waiting-k> K </waiting-k>
+                       <waiting-task-conf> Task </waiting-task-conf>
         </waiting-task> => .)
+  [futureUnblock]
+
+/* Basic semantics for await
+*/
+
+  // Sequencing ands
+  rule await(E1 && E2); => await(E1); ~> await(E2);
+  // reducing trues 
+  rule await(true); => .
+  // transforming other conditions in sets of disjunctive guards
+  rule await(E ?); => waiting(<guard> SetItem(E ?)</guard>)
+  rule await(wait); => waiting(<guard> SetItem(wait)</guard>)
+  rule await(E1 || E2); => waiting(<guard> SetItem(E1) SetItem(E2)</guard>)
+
+  syntax KItem ::= waiting(Bag)
+  // evaluating future expressions to future references
+  context waiting(<guard> _:Set SetItem(HOLE ?) </guard>)
```

Comment:
nu ar trebui "<guard> _:Set SetItem(HOLE ?)  _:Set </guard>" in loc de "<guard> _:Set SetItem(HOLE ?) </guard>"?